src/hooks/storage.ts

```diff
@@ -0,0 +1,30 @@
+import { Options } from 'got-cjs/dist/source';
+
+class Storage {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    storage: WeakMap<object, any>;
+
+    constructor() {
+        this.storage = new WeakMap();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    get<T extends object>(token: object | undefined): T | undefined {
+        if (!token) {
+            return;
+        }
+
+        if (!this.storage.has(token)) {
+            this.storage.set(token, {});
+        }
+
+        return this.storage.get(token);
+    }
+}
+
+const storage = new Storage();
+
+export const sessionDataHook = (options: Options): void => {
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    options.context.sessionData = storage.get(options.context.sessionToken as object | undefined);
```

Comment:
Given the number of interfaces we have for all the different properties of session data, I think it would make sense to have a proper `SessionData` interface.

Reply:
I did this move on purpose so each type can be stored in its corresponding file. I don't like putting everything into one file. I can go with this approach though. Hm?

Reply:
I'll leave this to you. I don't know much about TypeScript conventions. Or maybe @B4nan @vladfrangu can chime in?

Reply:
> I don't like putting everything into one file.

Don't look in the apify-client then 😅 

However, if they all point to the same interface, I'm pretty sure you can export the same name for an interface across all files, and TS can merge them? I haven't tried that if I'll be honest, but I know it works with augmentations 👀 

Either way, you shouldn't cast the object as `object` but rather `Record<string, unknown>` or `any/unknown/never` if it's not known yet.. At least that's what I do